From 544137953ce4434408caee2b817b225358d10507 Mon Sep 17 00:00:00 2001
From: tropf <tropf@noreply.codeberg.org>
Date: Tue, 8 Aug 2023 19:57:40 +0200
Subject: [PATCH 1/9] silence warnings

Some of the python methods create deprecation warnings. Filtering via an
environment variable did not work from a wrapper, hence this patch
silences all warnings by force.
---
 inkstitch.py                 | 4 ++++
 lib/api/server.py            | 3 +++
 lib/api/simulator.py         | 3 +++
 lib/api/stitch_plan.py       | 3 +++
 lib/extensions/print_pdf.py  | 3 +++
 lib/stitches/contour_fill.py | 7 ++++++-
 lib/utils/json.py            | 2 ++
 7 files changed, 24 insertions(+), 1 deletion(-)

diff --git a/inkstitch.py b/inkstitch.py
index 1dc5a3e3..a69caa0c 100644
--- a/inkstitch.py
+++ b/inkstitch.py
@@ -10,6 +10,10 @@ import sys
 import traceback
 from argparse import ArgumentParser
 from io import StringIO
+import warnings
+
+# ignore all warnings
+warnings.filterwarnings('ignore')
 
 if getattr(sys, 'frozen', None) is None:
     # When running in development mode, we want to use the inkex installed by
diff --git a/lib/api/server.py b/lib/api/server.py
index 26efa521..5cdf5475 100644
--- a/lib/api/server.py
+++ b/lib/api/server.py
@@ -24,6 +24,9 @@ from .lang import languages
 # this for electron axios
 from flask_cors import CORS
 
+import warnings
+warnings.filterwarnings('ignore')
+
 
 class APIServer(Thread):
     def __init__(self, *args, **kwargs):
diff --git a/lib/api/simulator.py b/lib/api/simulator.py
index 26c0246c..913cde9d 100644
--- a/lib/api/simulator.py
+++ b/lib/api/simulator.py
@@ -5,4 +5,7 @@
 
 from flask import Blueprint
 
+import warnings
+warnings.filterwarnings('ignore')
+
 simulator = Blueprint('simulator', __name__)
diff --git a/lib/api/stitch_plan.py b/lib/api/stitch_plan.py
index c70efd98..946bfd2c 100644
--- a/lib/api/stitch_plan.py
+++ b/lib/api/stitch_plan.py
@@ -7,6 +7,9 @@ from flask import Blueprint, g, jsonify
 
 from ..stitch_plan import stitch_groups_to_stitch_plan
 
+import warnings
+warnings.filterwarnings('ignore')
+
 
 stitch_plan = Blueprint('stitch_plan', __name__)
 
diff --git a/lib/extensions/print_pdf.py b/lib/extensions/print_pdf.py
index 1d5af76a..fbfabdf7 100644
--- a/lib/extensions/print_pdf.py
+++ b/lib/extensions/print_pdf.py
@@ -29,6 +29,9 @@ from ..svg.tags import INKSCAPE_GROUPMODE
 from ..threads import ThreadCatalog
 from .base import InkstitchExtension
 
+import warnings
+warnings.filterwarnings('ignore')
+
 
 def datetimeformat(value, format='%Y/%m/%d'):
     return value.strftime(format)
diff --git a/lib/stitches/contour_fill.py b/lib/stitches/contour_fill.py
index 2ea61afb..57ac3ae9 100644
--- a/lib/stitches/contour_fill.py
+++ b/lib/stitches/contour_fill.py
@@ -3,7 +3,12 @@ from itertools import chain
 
 import networkx as nx
 import numpy as np
-import trimesh
+
+import warnings
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
+    import trimesh
+
 from shapely.geometry import (GeometryCollection, LineString, MultiPolygon,
                               Point, Polygon)
 from shapely.geometry.polygon import orient
diff --git a/lib/utils/json.py b/lib/utils/json.py
index 22107c48..895f2d84 100644
--- a/lib/utils/json.py
+++ b/lib/utils/json.py
@@ -6,6 +6,8 @@
 from flask.json.provider import DefaultJSONProvider
 from ..exceptions import InkstitchException
 
+import warnings
+warnings.filterwarnings('ignore')
 
 class InkstitchJSONException(InkstitchException):
     pass
-- 
2.40.1

